browse/controllers/abs_page/__init__.py

```diff
@@ -189,50 +187,51 @@ def _check_request_headers(docmeta: DocMetadata,
         # If there is a more recent trackback ping, use that datetime
         last_mod_dt = response_data['trackback_ping_latest']
 
-    # Check for request headers If-Modified-Since and If-None-Match and compare
-    # them to the last modified time to determine whether we will return a
-    # "not modified" response
-    mod_since_dt = _time_header_parse(headers, 'If-Modified-Since')
-    none_match_dt = _time_header_parse(headers, 'If-None-Match')
-    not_modified = _not_modified(last_mod_dt, mod_since_dt, none_match_dt)
-
     last_mod_mime = mime_header_date(last_mod_dt)
-    headers['Last-Modified'] = last_mod_mime
-    headers['ETag'] = last_mod_mime
-    headers['Expires'] = abs_expires_header()[1]
+    etag = f'"{last_mod_mime}"'
+    
+    resp_headers['Last-Modified'] = last_mod_mime
+    resp_headers['ETag'] = etag
+    resp_headers['Expires'] = abs_expires_header()[1]
+    
+    not_modified = _not_modified(last_mod_dt,
+                                 _time_header_parse('If-Modified-Since'),
+                                 _get_req_header('if-none-match'),
+                                 etag)
 
     return not_modified
 
 
 def _not_modified(last_mod_dt: datetime,
                   mod_since_dt: Optional[datetime],
-                  none_match_dt: Optional[datetime])->bool:
-    if mod_since_dt and none_match_dt:
-        not_modified = (mod_since_dt >= last_mod_dt
-                        and none_match_dt >= last_mod_dt)
-    elif mod_since_dt and not none_match_dt:
-        not_modified = mod_since_dt >= last_mod_dt
-    elif none_match_dt and not mod_since_dt:
-        not_modified = none_match_dt >= last_mod_dt
+                  none_match: Optional[str],
+                  current_etag: str) -> bool:
+    if none_match and none_match == current_etag:
```

Comment:
I changed this logic a bunch and put the etag first because it is suppose to take priority over last-modified.